src/pages/Florist/index.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+  Button,
+  createStyles,
+  Grid,
+  makeStyles,
+  Theme,
+  Typography,
+} from "@material-ui/core";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { Redirect } from "react-router-dom";
+import { LoadingWrapper } from "../../components/LoadingWrapper";
+import { buyFlower } from "../../helpers/api/flowers/buyFlower";
+import { getFlowers } from "../../helpers/api/flowers/getFlowers";
+import { useUserState } from "../../store/user/useUserState";
+import { useApi } from "../../utils/api/useApi";
+import QuestionMark from "./assets/questionMark.png";
+import styles from "./Florist.module.css";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    container: {
+      width: "90%",
+      marginLeft: "2.5%",
+      padding: "5%",
+    },
+    header: {
+      margin: "2%",
+      marginLeft: "5%",
+      width: "100%",
+    },
+    flowerList: {
+      width: "100%",
+    },
+    card: {
+      margin: "1%",
+      height: "30%",
+      width: "30%",
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "center",
+    },
+    cardBought: {
+      margin: "1%",
+      height: "30%",
+      width: "30%",
+      display: "flex",
+      flexDirection: "column",
+      alignItems: "center",
+      backgroundColor: "#6ac69780",
+    },
+    buyButton: {
+      marginBottom: "0.125rem",
+      width: "100%",
+    },
+    errorMsg: {
+      backgroundColor: theme.palette.error.light,
+    },
+  })
+);
+
+export const Florist = () => {
+  const classes = useStyles();
+  const { userData, setUserData } = useUserState();
+  const [lastBought, setLastBought] = useState<string | null>(null);
+  const [flowerPrice, setFlowerPrice] = useState(0);
```

Comment:
Added flowerPrice state to keep track of current buying flower Price to use when updating global state balance